conans/client/generators/json_generator.py

```diff
@@ -0,0 +1,36 @@
+import json
+
+from conans.model import Generator
+
+
+def serialize_cpp_info(cpp_info):
+    keys = [
+        "version",
+        "description",
+        "rootpath",
+        "sysroot",
+        "include_paths", "lib_paths", "bin_paths", "build_paths", "res_paths",
+        "libs",
+        "defines", "cflags", "cppflags", "sharedlinkflags", "exelinkflags",
+    ]
+    res = {}
+    for key in keys:
+        res[key] = getattr(cpp_info, key)
+    return res
+
+
+class JsonGenerator(Generator):
+    @property
+    def filename(self):
+        return "conaninfo.json"
+
+    @property
+    def content(self):
+        info = {}
+        info["dependencies"] = []
+        for depname, cpp_info in self.deps_build_info.dependencies:
```

Comment:
For completitude I would like to see also here the values of `self.deps_env_info`, `self.env_info` and `self._deps_user_info`

Reply:
Done for `deps_env_info` and `deps_user_info`.
I did not manage to figure out when `self.env_info` is not None. Do you have some pointers to help me ?

Reply:
It is environmet information declared in the conanfile that receives the ``$ conan` install`` command. Check for info in the docs: http://docs.conan.io/en/latest/reference/conanfile/methods.html#env-info

Reply:
Hum. I'm missing something then. In the test I wrote I have:

```python
class HelloConan(ConanFile):
    def package_info(self):
        self.env_info.MY_VAR = "foo"
```

But inside the JsonGenerator, `self.conanfile.env_info` is still None.

I do find the 'MY_VAR' key in `self.deps_env_info.vars`, though

Reply:
You are right, deps_env_info is where env_info vars are propagated when doing the install: https://github.com/conan-io/conan/blob/207e2c70938b6e74c2279f827d51ec244d701be0/conans/client/installer.py#L453

So I think it is ok the way it is. Sorry for that

Reply:
No problem :)